cmd/sonobuoy/app/images.go

```diff
@@ -22,46 +22,102 @@ import (
 
-type imagesFlags struct {
-	e2eRegistryConfig string
-	plugin            string
-	kubeconfig        Kubeconfig
-
-	imagesflags *pflag.FlagSet
-}
+const numDockerRetries = 1
+)
 
-func ImagesFlagSet(cfg *imagesFlags) *pflag.FlagSet {
-	flagset := pflag.NewFlagSet("images", pflag.ExitOnError)
-	AddPluginFlag(&cfg.plugin, flagset)
-	AddKubeconfigFlag(&cfg.kubeconfig, flagset)
-	cfg.imagesflags = flagset
-	return flagset
+type imagesFlags struct {
+	e2eRegistryConfig  string
+	plugin             string
+	username           string
+	password           string
+	imagesSaveFileName string
+	downloadTar        bool
+	kubeconfig         Kubeconfig
 }
 
 func NewCmdImages() *cobra.Command {
+	// Main command
 	cmd := &cobra.Command{
 		Use:   "images",
 		Short: "Manage images used in a plugin. Supported plugins are: 'e2e'",
-		Run:   getImages,
+		Run:   listImages,
 		Args:  cobra.ExactArgs(0),
 	}
 
-	cmd.Flags().AddFlagSet(ImagesFlagSet(&imagesflags))
+	AddKubeconfigFlag(&imagesflags.kubeconfig, cmd.Flags())
+	AddPluginFlag(&imagesflags.plugin, cmd.Flags())
+
+	// Pull command
+	pullCmd := &cobra.Command{
+		Use:   "pull",
+		Short: "Pulls images to local docker client for a specific plugin",
+		Run:   pullImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	AddKubeconfigFlag(&imagesflags.kubeconfig, pullCmd.Flags())
+	AddPluginFlag(&imagesflags.plugin, pullCmd.Flags())
+
+	// Download command
+	downloadCmd := &cobra.Command{
+		Use:   "download",
+		Short: "Saves downloaded images from local docker client to a tar file",
+		Run:   downloadImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	AddKubeconfigFlag(&imagesflags.kubeconfig, downloadCmd.Flags())
+	AddPluginFlag(&imagesflags.plugin, downloadCmd.Flags())
+
+	// Push command
+	pushCmd := &cobra.Command{
+		Use:   "push",
+		Short: "Pushes images to docker registry for a specific plugin",
+		Run:   pushImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	AddE2ERegistryConfigFlag(&imagesflags.e2eRegistryConfig, pushCmd.Flags())
+	AddKubeconfigFlag(&imagesflags.kubeconfig, pushCmd.Flags())
+	AddPluginFlag(&imagesflags.plugin, pushCmd.Flags())
+	pushCmd.MarkFlagRequired(e2eRegistryConfigFlag)
+
+	// Delete command
+	deleteCmd := &cobra.Command{
+		Use:   "delete",
+		Short: "Deletes all images downloaded to local docker client",
+		Run:   deleteImages,
+		Args:  cobra.ExactArgs(0),
+	}
+	AddE2ERegistryConfigFlag(&imagesflags.e2eRegistryConfig, deleteCmd.Flags())
+	AddKubeconfigFlag(&imagesflags.kubeconfig, deleteCmd.Flags())
+	AddPluginFlag(&imagesflags.plugin, deleteCmd.Flags())
+
+	cmd.AddCommand(pullCmd)
+	cmd.AddCommand(pushCmd)
+	cmd.AddCommand(downloadCmd)
+	cmd.AddCommand(deleteCmd)
+
 	return cmd
 }
 
-func getImages(cmd *cobra.Command, args []string) {
+func listImages(cmd *cobra.Command, args []string) {
 
 	switch imagesflags.plugin {
 	case "e2e":
 
+		if len(imagesflags.e2eRegistryConfig) > 0 {
+			// Check if the e2e file exists
+			if _, err := os.Stat(imagesflags.e2eRegistryConfig); os.IsNotExist(err) {
```

Comment:
Based on the err message you should just check if `err != nil` here; otherwise we're not catching any of the "cannot be opened" cases.

Reply:
This should check if the file doesn't exist or if it cannot be opened.